configs/bundle_config.yaml

```diff
@@ -141,6 +141,17 @@ databundles:
     - data/hydrobasins/hybas_world_lev04_v1c.shp
     - data/hydrobasins/hybas_world_lev05_v1c.shp
 
+  # global data for hydrobasins
+  bundle_hydrobasins:
+    countries: [Earth]
+    tutorial: false
+    category: common
+    destination: "data/hydrobasins"
+    urls:
+      direct: https://data.hydrosheds.org/file/HydroBASINS/standard/hybas_af_lev06_v1c.zip
```

Comment:
I comment here and below I add some examples.
We could define a new class like "urls: hydrobasins: " to distinguish the download procedure that is different than the others.
the new code "hydrobasins" require the definition of a new function like download_and_unzip_hydrobasins

Then, the url of the hydrobasin may be: "https://data.hydrosheds.org/file/HydroBASINS/standard/hybas_af_lev{:02d}_v1c.zip"

Inside the new download_and_unzip_hydrobasins, we need to do "url_hbasin = urls["hydrobasins"].forma({level from config file}"

With this format, if we add one bundle_hydrobasin for every region in the hydrobasin website, this workflow will automatically download all of them. Note that the output file of each databundle shall account for the region (e.g. AF) and shall be filled with the level value similarly to above.

Then, we need to merge them into a unique shape file. To do so, we can create a dedicated rule to do so: the function takes in input the list of hydro shape files, calculated with a dedicated function, and in output creates the merged file.
Alternatively the merging process can be included into the retrieve databundle, if that's easier to start with for you

Reply:
Absolutely agree that a dedicated download function would be helpful, and great point about the merging procedure!

Regarding a special treatment for different regions, not sure I get your point right. Currently, we have a single file of global coverage, and it seems to work pretty well. Why wouldn't we reproduce the same approach, regardless of the requested region? The global dataset is about 5MB which is ~200 time smaller as compared with our environment 🙃 

Do you mean that it doesn't look polite if we'd re-creating a chunk of data set locally by default? 🙂 In this case I do agree that it makes absolute sense to go for the approach you suggest.

Reply:
The size of the dataset depends on the level. I've done a scan and the global v12 level may be of 500MB zipped, but it is unlikely to be commonly used.
We can go for it :) I like the proposal!

However, unfortunately there is no global bundle in the hydrobasins, we need to manually download all regions and merge them regardless.

Reply:
Thanks :)

Mmm... Probably, it could make sense to add a warning if the data to be loaded exceed some 100MB.

No problem with creating a merged dataset :)